### DISCUSSION TITLE
Dirty fields object is empty when changing a non-input field component with array of object values

### DISCUSSION THREAD

**Discussion by Slmii** [react-hook-form/react-hook-form#6202] [Q&A]
Hello,

I want to retreive the `isDirty` and `dirtyFields` from the `formState`, however, when I change an input component the changed input's name shows in the `dirtyFields` object. When I change a non-input component, for example I have a list with checkboxes that is conncected to the formState, the `dirtyFields` does NOT include this. 

Changing an input field: 
![image](https://user-images.githubusercontent.com/9927849/128826392-3691e54b-8de2-4eb5-965e-3ab80ff0f4ef.png)

Checking/unchecking a list with checkboxes:
![image](https://user-images.githubusercontent.com/9927849/128826463-0c75ec03-9497-4d41-9874-ee0ae7bab22d.png)

It does say its dirty but the name does not show up there. This is how the values in the form (of the non-input component) looks like. It's an array of objets. Checking/unchecking adds/removes an object from the array.
![image](https://user-images.githubusercontent.com/9927849/128826677-9474d2ca-125d-423a-88eb-e743142d6d36.png)

It does seem to work fine with non-array objects (which is also a non-input component):
![image](https://user-images.githubusercontent.com/9927849/128827166-dc00528e-8c4d-4ff8-9cf7-c2a94a90cae5.png)

Is there a way to solve this?

**Comment by bluebill1049** [comment]
provide a codesandbox.

**Reply by Slmii** [reply]
https://codesandbox.io/s/restless-field-mdsx6?file=/src/App.tsx

When you make an `onChange` on the select it logs `isDirty` and `dirtyFields`. However it will only detect the dirty state of the input field. When you select/deselect the checkboxes it doesnt see the form value as dirty.

If you press the submit button you can see that items DO get added/removed from the form values.

**Reply by Slmii** [reply]
I also noticed that when I leave focus and focus out an input, without changing anything, `isDirty` becomes true. Is there a way to prevent this? Because my current solution depends purely on the `isDirty` value since the checkbox is not detected as dirty.

**Reply by bluebill1049** [reply]
check for your form values against the defaultValues that you supplied.

**Reply by Slmii** [reply]
isnt that just a workaround? shouldnt `dirtyFields` be working properly with this?

**Reply by bluebill1049** [reply]
No, I am asking you to check your form values against the defaultValues, see if they are matching or not. Dirty field is driven by the form values with the default value that you supplied, I can see missing value in your default values.

**Reply by Slmii** [reply]
in my case the default values will be always have empty default at init. after selecting an option from the dropdown the default values get filled (with `reset`). 

My default values and formvalues (after selecting an option from the dropdown) will never match. I want to open a popup (after re-selecting an option) if there are changes to the formValues that are loaded in dynamically after selecting an option.

```
1. default values at init (example)
{
    name: '',
    age: ''
}

2. select an option and fill defaultValues in form dynamically
{
    name: 'John Doe',
    age: '25'
}

3. select another option > check if there are changes in the form.
if I compare the defaultValues (which is always empty at init) and formValues here it'll never be a match and always see it as `dirty`
```
